src/dialogs/more-info/ha-more-info-dialog.ts

```diff
@@ -0,0 +1,298 @@
+import { customElement, LitElement, property, css } from "lit-element";
+import { html } from "lit-html";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import { getRecentWithCache, CacheConfig } from "../../data/cached-history";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { mdiClose, mdiSettings, mdiPencil } from "@mdi/js";
+
+const DOMAINS_NO_INFO = ["camera", "configurator", "history_graph"];
+const EDITABLE_DOMAINS_WITH_ID = ["scene", "automation"];
+const EDITABLE_DOMAINS = ["script"];
+
+@customElement("ha-more-info-dialog")
+export class MoreInfoDialog extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() private _stateHistory?: object;
+
+  @property({ type: Boolean, reflect: true }) public large = false;
+
+  @property() private _cacheConfig?: CacheConfig;
+
+  private _interval?: number;
+
+  protected updated(changedProperties) {
+    super.updated(changedProperties);
+    if (changedProperties.has("hass")) {
+      const oldHass = changedProperties.get("hass");
+      if (!oldHass || oldHass.moreInfoEntityId !== this.hass.moreInfoEntityId) {
+        if (this.hass.moreInfoEntityId) {
+          this.setAttribute(
+            "data-domain",
+            computeDomain(this.hass.moreInfoEntityId)
+          );
+          this.large = false;
+          this._stateHistory = undefined;
+          if (this._computeShowHistoryComponent(this.hass.moreInfoEntityId)) {
+            this._cacheConfig = {
+              refresh: 60,
+              cacheKey: `more_info.${this.hass.moreInfoEntityId}`,
+              hoursToShow: 24,
+            };
+            this._getStateHistory();
+            clearInterval(this._interval);
+            if (this._cacheConfig.refresh) {
+              this._interval = window.setInterval(() => {
+                this._getStateHistory();
+              }, this._cacheConfig.refresh * 1000);
+            }
+          }
+        } else {
+          this._stateHistory = undefined;
+          clearInterval(this._interval);
+          this._interval = undefined;
+        }
+      }
+    }
+  }
+
+  protected render() {
+    if (!this.hass.moreInfoEntityId) {
+      return html``;
+    }
+    const entityId = this.hass.moreInfoEntityId;
+    const stateObj = this.hass.states[entityId];
+    const domain = computeDomain(entityId);
+
+    return html`
+      <ha-dialog open @closed=${this._close} .heading=${true} hideActions>
+        <app-toolbar slot="heading">
+          <mwc-icon-button
+            .label=${this.hass.localize("ui.dialogs.more_info_control.dismiss")}
+            dialogAction="cancel"
+          >
+            <ha-svg-icon .path=${mdiClose}></ha-svg-icon>
+          </mwc-icon-button>
+          <div class="main-title" main-title @click=${this._enlarge}>
+            ${stateObj ? computeStateName(stateObj) : ""}
```

Comment:
Should we even render all of this when there is no `stateObj` ? I think we only did that in Polymer because it was so difficult to not do it.

Reply:
Should we no render the dialog at all?

Reply:
Nope. No state object means we hide the dialog.